app/assets/javascripts/code/services/_draw_session.js

```diff
@@ -61,7 +61,21 @@ export const drawSession = (
       return session.markers;
     },
 
-    calculateHeatLevel: function(value) {
+    drawMobileSessionStartPoint: function(session) {
```

Comment:
There's a lot of smelly things in here:
- `if(!session || !sensors.anySelected())` maybe the function shouldn't even be called in the first place in those cases?
- `session.markers = []` maybe this is not this module's responsibility (ie maybe a session module should be doing this?)
- `var markerOptions = {title: session.title, zIndex: 0};` this seems like a map responsibility
- `session.markers.push(map.drawMarker(lngLatObject, markerOptions, null, "0"));` mixed session / map responsibility?
- `session.drawed = true;` again session

The code is really procedural, maybe we could try to separate the responsibilities better. I know this is what we have in place and it's difficult to understand what's going on / how to decouple. But this is where we can make the difference